Source/SwiftLintFramework/Rules/ParametersListLengthRule.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  ParemeterListLengthRule.swift
+//  SwiftLint
+//
+//  Created by Denis Lebedev on 26/01/2016.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct ParametersListLengthRule: ASTRule, ConfigProviderRule {
+    public var config = SeverityLevelsConfig(warning: 6, error: 10)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "parameters_list_length",
+        name: "Parameter List Length",
+        description: "Length of parameter list should be short.",
+        nonTriggeringExamples: [
+            //"func f1(a: Int, b: Int) { foo(1,4) }\n" +
+            //"func f2(p1: Int, p2: Int) { f1(1, 2)}",
+            //"func f(a: Int, b: Int, c: Int, d: Int, f: Int, x: Int = 42) {}",
+        "extension File {\n" +
+        "private func invalidDocOffsets(dictionary: [String: SourceKitRepresentable]) -> [Int] {\n" +
+        "let substructure = (dictionary as? [SourceKitRepresentable])?.flatMap { $0 as? [String: SourceKitRepresentable] } ?? []}}"
+        ],
+        triggeringExamples: [
+            "func f(a: Int, b: Int, c: Int, d: Int, e: Int, f: Int) {}",
+        ]
+    )
+
+    public func validateFile(file: File, kind: SwiftDeclarationKind,
+        dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+            if !functionKinds.contains(kind) {
+                return []
+            }
+
+            let substructure = dictionary["key.substructure"] as? [SourceKitRepresentable] ?? []
+            let paramsCount = substructure.filter {
+                guard let subDict = $0 as? [String: SourceKitRepresentable],
+                    key = subDict["key.kind"] as? String  else {
+                        return false
+                }
+
+                return key == "source.lang.swift.decl.var.parameter"
```

Comment:
You could avoid the hardcoded string literal by doing `return SwiftDeclarationKind(rawValue: key) == . VarParameter` instead.